panic() declared with C++ linkage in pico-sdk/src/rp2_common/pico_lwip/include/arch/cc.h when building a C++ executable

I ran into this issue when building with the current develop branch (commit 605e21ae6) for mbedtls support.

`void panic(const char *fmt, ...);` at [arch/cc.h line 73](https://github.com/raspberrypi/pico-sdk/blob/605e21ae6f28cd35b5d4be454384684a409e7a67/src/rp2_common/pico_lwip/include/arch/cc.h#L73) is not declared with C linkage when compiling for C++, resulting in the following error:

```
... pico-sdk/src/rp2_common/pico_platform/include/pico/platform.h:315:32: error: conflicting declaration of 'void panic(const char*, ...)' with 'C' linkage
315 | void __attribute__((noreturn)) panic(const char *fmt, ...);
    |                                ^~~~~
... pico-sdk/src/rp2_common/pico_lwip/include/arch/cc.h:73:6: note: previous declaration with 'C++' linkage
 73 | void panic(const char *fmt, ...);
    |      ^~~~~
```

Surrounding the declaration in `arch/cc.h` with
```c++
#ifdef __cplusplus
extern "C" {
#endif
void panic(const char *fmt, ...);
#ifdef __cplusplus
}
#endif
```
fixes the issue for me.

--- a/src/rp2_common/pico_lwip/include/arch/cc.h
+++ b/src/rp2_common/pico_lwip/include/arch/cc.h
@@ -70,7 +70,13 @@ typedef int sys_prot_t;
 #endif
 
 #ifndef LWIP_PLATFORM_ASSERT
+#ifdef __cplusplus
+extern "C" {
+#endif
 void panic(const char *fmt, ...);
+#ifdef __cplusplus
+}
+#endif
 #define LWIP_PLATFORM_ASSERT(x) panic(x)
 #endif
 

